kernel/kernel-api/src/main/java/io/delta/kernel/hook/PostCommitHook.java

```diff
@@ -19,8 +19,17 @@
 import io.delta.kernel.engine.Engine;
 import java.io.IOException;
 
+/**
+ * A hook for executing operation after a transaction commit. Hooks are added in the Transaction and
+ * engine need to invoke the hook explicitly for executing the operation.
+ */
 public interface PostCommitHook {
 
+  enum PostCommitHookType {
+    // Write a new checkpoint at the version committed by the txn if required.
```

Comment:
this is maybe a little ambiguous as to when it is present (i.e. is it always present and only sometime checkpoints?)

Maybe something like "Write a new checkpoint at the version committed by the txn. This hook is present when the table is ready for checkpoint according to its configured checkpoint interval"

Reply:
You can do `/* */` comment for the enum